example/include/ExampleApp.h

```diff
@@ -0,0 +1,78 @@
+
+/*
+ * This example is explained as part of the \ref example documentation page. Please refere there for step-by-step
+ * explanations. Reading the full example might not be a good starting point for learning ApplicationCore as it can
+ * be overwelming and lacks important background information.
+ */
+
+#include "AverageCurrent.h"
+#include "Controller.h"
+#include "SetpointRamp.h"
+
+#include <ChimeraTK/ApplicationCore/ApplicationCore.h>
+#include <ChimeraTK/ApplicationCore/ConfigReader.h>
+#include <ChimeraTK/ApplicationCore/PeriodicTrigger.h>
+#include <ChimeraTK/ApplicationCore/ScriptedInitialisationHandler.h>
+
+namespace ctk = ChimeraTK;
+
+//! [Class Definition Start]
+class ExampleApp : public ctk::Application {
+ public:
+  using ctk::Application::Application;
+  ~ExampleApp() override;
+
+ private:
+  //! [Class Definition Start]
+  // Set the name of the DMAP file to define the devices. Must be done before instantiating any DeviceModule.
+  // Using the application name as a base helps for automated testing against different config files.
+  //! [SetDMapFilePath]
+  ctk::SetDMapFilePath dmapPath{getName() + ".dmap"};
+  //! [SetDMapFilePath]
+
+  // Provide configuration constants as process variables and for use at construction phase
+  // Using the application name as a base helps for automated testing against different config files.
+  ctk::ConfigReader config{this, "/", getName() + "-config.xml"};
+
+  // Periodic trigger used to readout data from the device periodically.
+  //! [PeriodicTrigger Instance]
+  ctk::PeriodicTrigger timer{this, "Timer", "Periodic timer for the controller"};
+  //! [PeriodicTrigger Instance]
+
+  // Publish the content of the device "oven" defined in the DMAP file to the control system and to the application
+  // modules. The "tick" output of the PeriodicTimer "Timer" defined above is used as a readout trigger (for all
+  // poll-typed device registers).
+  //! [Device]
+  ctk::DeviceModule oven{this, "oven", "/Timer/tick"};
+  //! [Device]
+
+  // Initialisation handler: execute Python script to initialise oven device
+  //! [ScriptedInitHandler]
+  ctk::ScriptedInitHandler ovenInit{this, "ovenInit", "Initialisation of oven device", "./ovenInit.py", oven};
+  //! [ScriptedInitHandler]
+
+  //! [Controller Instance]
+  // Instantiate the temperature controller module
+  Controller controller{this, "Controller", "The temperature controller"};
+  //! [Controller Instance]
+
+  //! [AverageCurrent Instance]
+  // Instantiate the heater current averaging module
+  AverageCurrent averageCurrent{this, "AverageCurrent", "Provide averaged heater current"};
+  //! [AverageCurrent Instance]
+
+  // Optionally instantiate some module(s) for automation
+  class Automation : public ctk::ModuleGroup {
+    using ctk::ModuleGroup::ModuleGroup;
+
+    SetpointRamp ramp{this, "SetpointRamp", "Slow ramping of temperator setpoint"};
+  };
+  Automation automation{config.get<int>("Configuration/enableAutomation") ?
```

Comment:
Does this trick require a module group? It's quite some lines of code to define it. If it's not optional couldn't I just write

Reply:
No it does not require a module group, you can make the module itself optional, too. The ModuleGroup and the optional instantiation have nothing to do with each other. Should we show this separately? That would require yet another ApplicationModule though...

Reply:
The question is: how much do we want to show? And how realistic and full-blown can we get without overwhelming the reader.

What struck me odd: A ModuleGroup with a single module does not make sense any more since you can give the hierarchy in the name. A ModuleGroup should at least have two modules inside.

However, it makes perfect sense to show how to instantiate optionally with a module group, because you want to turn on and off a larger section without doing every module individually.

Let's leave the code as is. Maybe put a comment that you usually have multiple modules per group, and we only show one to keep it simple (only show the concept and the syntax).